src/game/common/system/geometry.cpp

```diff
@@ -270,30 +279,37 @@ void GeometryInfo::Get_2D_Bounds(const Coord3D &pos, float angle, Region2D &regi
             region.hi.y = pos.y + m_majorRadius;
             break;
         case GEOMETRY_BOX: {
-            float sin_theta = Sin(angle);
             float cos_theta = Cos(angle);
+            float sin_theta = Sin(angle);
+
             float adj_major = cos_theta * m_majorRadius;
             float adj_minor = cos_theta * m_minorRadius;
+
             float opp_major = sin_theta * m_majorRadius;
             float opp_minor = sin_theta * m_minorRadius;
-            region.lo.x = region.hi.x = float(pos.x - adj_major) - opp_minor;
-            region.lo.y = region.hi.y = float(pos.y + adj_minor) - opp_major;
 
-            float tmp_x = float(adj_major + pos.x) - opp_minor;
-            float tmp_y = float(opp_major + pos.y) + adj_minor;
+            float tmp_x = float(pos.x - adj_major) - opp_minor;
+            float tmp_y = float(pos.y + adj_minor) - opp_major;
+            region.lo.x = tmp_x;
+            region.lo.y = tmp_y;
+            region.hi.x = tmp_x;
+            region.hi.y = tmp_y;
+
+            tmp_x = float(pos.x + adj_major) - opp_minor;
+            tmp_y = float(pos.y + adj_minor) + opp_major;
             region.lo.x = std::min(region.lo.x, tmp_x);
             region.lo.y = std::min(region.lo.y, tmp_y);
             region.hi.x = std::max(region.hi.x, tmp_x);
             region.hi.y = std::max(region.hi.y, tmp_y);
 
-            tmp_x = float(opp_minor + pos.x) + adj_major;
+            tmp_x = float(pos.x + adj_major) + opp_minor;
             tmp_y = float(pos.y - adj_minor) + opp_major;
             region.lo.x = std::min(region.lo.x, tmp_x);
             region.lo.y = std::min(region.lo.y, tmp_y);
             region.hi.x = std::max(region.hi.x, tmp_x);
             region.hi.y = std::max(region.hi.y, tmp_y);
 
-            tmp_x = float(opp_minor + pos.x) - adj_major;
+            tmp_x = float(pos.x - adj_major) + opp_minor;
             tmp_y = float(pos.y - adj_minor) - opp_major;
```

Comment:
I see this function as calculating the 4 x points and the 4 y points and using the min/max of the set.
```cpp
float points_x[4] = {
    pos.x - adj_major - opp_minor, 
    pos.x + adj_major - opp_minor, 
    pos.x + adj_major + opp_minor,
    pos.x - adj_major + opp_minor,
};
region.lo.x = std::min_element(std::begin(points_x), std::end(points_x));
region.hi.x = std::max_element(std::begin(points_x), std::end(points_x));
// repeat for y
```
And when you have it like that you can see that pos.x is common across this so you could remove that from the table calc and perform the additon only once! Simplifying the code, making it even more readable and speeding it up by reducing calcs and branches. But perhaps a bit too radicle.